system-parachains/asset-hubs/asset-hub-polkadot/src/staking/mod.rs

```diff
@@ -68,5 +150,291 @@ impl pallet_delegated_staking::Config for Runtime {
+	pub MinerTxPriority: TransactionPriority = TransactionPriority::max_value() / 2;
+}
+
+impl multi_block::unsigned::Config for Runtime {
+	type MinerPages = MinerPages;
+	type OffchainStorage = ConstBool<true>;
+	type OffchainSolver = SequentialPhragmen<AccountId, SolutionAccuracyOf<Runtime>>;
+	type MinerTxPriority = MinerTxPriority;
+	type OffchainRepeat = OffchainRepeat;
+	type WeightInfo = multi_block::weights::polkadot::MultiBlockUnsignedWeightInfo<Self>;
+}
+
+parameter_types! {
+	/// Miner transaction can fill up to 75% of the block size.
+	pub MinerMaxLength: u32 = Perbill::from_rational(75u32, 100) *
+		*RuntimeBlockLength::get()
+		.max
+		.get(DispatchClass::Normal);
+}
+
+impl multi_block::unsigned::miner::MinerConfig for Runtime {
+	type AccountId = AccountId;
+	type Hash = Hash;
+	type MaxBackersPerWinner = <Self as multi_block::verifier::Config>::MaxBackersPerWinner;
+	type MaxBackersPerWinnerFinal =
+		<Self as multi_block::verifier::Config>::MaxBackersPerWinnerFinal;
+	type MaxWinnersPerPage = <Self as multi_block::verifier::Config>::MaxWinnersPerPage;
+	type MaxVotesPerVoter =
+		<<Self as multi_block::Config>::DataProvider as ElectionDataProvider>::MaxVotesPerVoter;
+	type MaxLength = MinerMaxLength;
+	type Solver = <Runtime as multi_block::unsigned::Config>::OffchainSolver;
+	type Pages = Pages;
+	type Solution = NposCompactSolution16;
+	type VoterSnapshotPerBlock = <Runtime as multi_block::Config>::VoterSnapshotPerBlock;
+	type TargetSnapshotPerBlock = <Runtime as multi_block::Config>::TargetSnapshotPerBlock;
+}
+
+// FAIL-CI wrong use the one from the relay
+pub struct EraPayout;
+impl pallet_staking_async::EraPayout<Balance> for EraPayout {
+	fn era_payout(
+		_total_staked: Balance,
+		_total_issuance: Balance,
+		era_duration_millis: u64,
+	) -> (Balance, Balance) {
+		const MILLISECONDS_PER_YEAR: u64 = (1000 * 3600 * 24 * 36525) / 100;
+		// A normal-sized era will have 1 / 365.25 here:
+		let relative_era_len =
+			FixedU128::from_rational(era_duration_millis.into(), MILLISECONDS_PER_YEAR.into());
+
+		// Fixed total TI that we use as baseline for the issuance.
+		let fixed_total_issuance: i128 = 5_216_342_402_773_185_773;
+		let fixed_inflation_rate = FixedU128::from_rational(8, 100);
+		let yearly_emission = fixed_inflation_rate.saturating_mul_int(fixed_total_issuance);
+
+		let era_emission = relative_era_len.saturating_mul_int(yearly_emission);
+		// 15% to treasury, as per Polkadot ref 1139.
+		let to_treasury = FixedU128::from_rational(15, 100).saturating_mul_int(era_emission);
+		let to_stakers = era_emission.saturating_sub(to_treasury);
+
+		(to_stakers.saturated_into(), to_treasury.saturated_into())
+	}
+}
+
+parameter_types! {
+	// Six sessions in an era (6 hours).
+	pub const SessionsPerEra: SessionIndex = prod_or_fast!(6, 2);
+	/// Duration of a relay session in our blocks. Needs to be hardcoded per-runtime.
+	pub const RelaySessionDuration: BlockNumber = 1 * HOURS;
```

Comment:
Did you also check these values @kianenigma? I did not put TODOs everywhere, but the stuff is all from Westend.